tests/cpp/pattern_match_test.cc

```diff
@@ -0,0 +1,112 @@
+#include <gtest/gtest.h>
+#include "../src/arithmetic/pattern_match.h"
+
+TEST(Pattern, Basic) {
+  using namespace tvm;
+  using namespace tvm::arith;
+  Var x("x"), y("y"), z("z");
+  arith::PVar<Expr> px, py, pz;
+  arith::PVar<Type> pt;
+  arith::PVar<int> planes;
+
+  // arithmetics
+  auto r = 1 + (y + 1);
+  CHECK(!(px + (px + px)).Match(r));
+  CHECK(!(px + (py + py)).Match(r));
+  CHECK((px + (py + pz)).Match(r));
+  {
```

Comment:
Does it make sense to make changes such that `(px+(px+py))` too matches `1+ (y+1)`, that is, to take into consideration the `+`'s commutativity? Or is that commutativity something that needs to be taken care of independently?